File: backend/sportsnet/src/main/java/com/sports/sportsnet/entity/PersonalProfile.java
package com.sports.sportsnet.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EqualsAndHashCode(of="id")
@ToString(exclude = "user")
public class PersonalProfile {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(updatable = false, nullable = false)
    private UUID id;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false, unique = true)
    private User user;

    private LocalDate dateOfBirth;
    private Double heightCm;
    private Double weightKg;
    private String country;
    private String location;
    private String profileImagePath; // stores file name only

    @Column(nullable = false)
    private boolean contactVisible;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
    }
}

File: frontend/sportsnet-frontend/src/types/personalProfile.ts
export interface PersonalProfileRequest {
  dateOfBirth?: string | null; // "YYYY-MM-DD" or null
  heightCm?: number | null;
  weightKg?: number | null;
  country?: string | null;
  location?: string | null;
  contactVisible: boolean;
}

export interface PersonalProfileResponse {
  userId: string;
  fullName: string;
  age: number | null;
  heightCm: number | null;
  weightKg: number | null;
  country: string | null;
  location: string | null;
  profileImageUrl: string | null;
  telephone: string | null;
  email: string | null;
}
